src/freenet/clients/http/staticfiles/js/checkall.js

```diff
@@ -1,9 +1,13 @@
-// copypaste javascript from http://stackoverflow.com/a/7251104/7666
+// @license magnet:?xt=urn:btih:1f739d935676111cfff4b4693e3816e664797050&dn=gpl-3.0.txt GPL-v3-or-Later
 function checkAll(bx, classname) {
-  var cbs = document.getElementsByClassName(classname)[0].getElementsByTagName("input");
-  for(var i=0; i < cbs.length; i++) {
-    if(cbs[i].type == "checkbox") {
-      cbs[i].checked = bx.checked;
+  var matching = document.getElementsByClassName(classname);
+  for (var matchingElement of matching) {
```

Comment:
Yes, the changed behavior now checks all elements with the given class, not just the first. For multiple tables on the same page, that can check all checkboxes of all tables, but since the submit button only takes the correct form into account, this leads to correct behavior.

The previous behavior made the checkall box ineffective for the second table on a page.